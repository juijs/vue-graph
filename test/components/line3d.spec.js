import { mount } from '@vue/test-utils'
import Graph3DLine from '@/components/line3d'

describe('line3d.js', () => {
    it('snapshot', () => {
        const wrapper = mount(Graph3DLine, {
            props: {
                width: 800,
                height: 600,
                axisMin: 0,
                axisMax: 50,
                paddingTop: 100,
                paddingBottom: 100,
                depth: 180,
                names: [ 'MS', 'Apple', 'Google' ],
                labels: [ '1Q', '2Q', '3Q', '4Q' ],
                values: [
                    [ 0, 15, 8, 18 ],
                    [ 15, 6, 10, 5 ],
                    [ 20, 20, 20, 20 ]
                ]
            }
        });

        wrapper.vm.$nextTick(() => {
            expect(wrapper.vm.$el).toMatchSnapshot();
        });
    });
});