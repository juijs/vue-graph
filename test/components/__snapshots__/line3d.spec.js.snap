// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`line3d.js snapshot 1`] = `
<div
  style="position: relative; user-select: none; webkit-user-select: none;"
  unselectable="on"
>
  <svg
    buffered-rendering="dynamic"
    height="600"
    style="font-family:arial,Tahoma,verdana,classic;background:#fff"
    width="800"
  >
    <g
      transform="translate(0.5,0.5)"
    >
      <defs>
        <clippath
          id="axis-clip-id-0.0"
        >
          <rect
            height="400"
            width="700"
            x="0"
            y="0"
          />
        </clippath>
        <clippath
          id="axis-clip-rect-id-0"
        >
          <rect
            height="400"
            width="700"
            x="0"
            y="0"
          />
        </clippath>
      </defs>
      <rect
        fill="#fff"
        fill-opacity="0"
        height="400"
        rx="0"
        ry="0"
        stroke="#fff"
        stroke-width="0"
        transform="translate(50,100)"
        width="700"
        x="0"
        y="0"
      />
      <g
        class="grid-block"
        transform="translate(50,100)"
      >
        <g>
          <text
            dx="3.6666666666666665"
            dy="0"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            x="104.41773986816406"
            y="360.97052001953125"
          >
            1Q
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="173.58001708984375"
            x2="240.40103149414062"
            y1="367.6874084472656"
            y2="282.80914306640625"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="240.40103149414062"
            x2="225.12510681152344"
            y1="282.80914306640625"
            y2="-29.279813766479492"
          />
          <text
            dx="3.6666666666666665"
            dy="0"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            x="246.6279754638672"
            y="393.7328796386719"
          >
            2Q
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="321.27191162109375"
            x2="376.1514892578125"
            y1="401.1138916015625"
            y2="311.2255859375"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="376.1514892578125"
            x2="379.6694030761719"
            y1="311.2255859375"
            y2="-7.703784942626953"
          />
          <text
            dx="3.6666666666666665"
            dy="0"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            x="398.5799560546875"
            y="428.26806640625"
          >
            3Q
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="478.70556640625"
            x2="521.6436767578125"
            y1="436.3132019042969"
            y2="341.41485595703125"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="521.6436767578125"
            x2="543.9554443359375"
            y1="341.41485595703125"
            y2="15.645094871520996"
          />
          <text
            dx="3.6666666666666665"
            dy="0"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="middle"
            x="560.273681640625"
            y="464.5761413574219"
          >
            4Q
          </text>
        </g>
        <g>
          <polygon
            fill="#dcdcdc"
            fill-opacity="0.3"
            points="35.62984085083008,336.0338134765625 645.8809204101562,473.2853698730469 676.8776245117188,373.3769836425781 114.3923110961914,256.1655578613281 35.62984085083008,336.0338134765625"
            stroke="#bfbfbf"
            stroke-opacity="1"
            stroke-width="2"
          />
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="645.8809204101562"
            x2="676.8776245117188"
            y1="473.2853698730469"
            y2="373.3769836425781"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="676.8776245117188"
            x2="717.9832153320312"
            y1="373.3769836425781"
            y2="40.766822814941406"
          />
        </g>
      </g>
      <g
        class="grid-range"
        transform="translate(50,100)"
      >
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="-5.475783348083496"
            x2="80.3225326538086"
            y1="11.45443344116211"
            y2="-49.08299255371094"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="80.3225326538086"
            x2="717.9832153320312"
            y1="-49.08299255371094"
            y2="40.766822814941406"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="-12.408380508422852"
            y="10.36248779296875"
          >
            50
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="-1.3652335405349731"
            x2="83.72949981689453"
            y1="47.475440979003906"
            y2="-14.995070457458496"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="83.72949981689453"
            x2="713.8726806640625"
            y1="-14.995070457458496"
            y2="77.5909194946289"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="-8.21191692352295"
            y="46.35222625732422"
          >
            45
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="2.7453370094299316"
            x2="87.13648223876953"
            y1="82.70466613769531"
            y2="18.30107307434082"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="87.13648223876953"
            x2="709.7620849609375"
            y1="18.30107307434082"
            y2="113.62322235107422"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="-4.015430450439453"
            y="81.55017852783203"
          >
            40
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="6.855886936187744"
            x2="90.54346466064453"
            y1="117.14208221435547"
            y2="50.805419921875"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="90.54346466064453"
            x2="705.6515502929688"
            y1="50.805419921875"
            y2="148.8637237548828"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="0.18105493485927582"
            y="115.95633697509766"
          >
            35
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="10.96645736694336"
            x2="93.950439453125"
            y1="150.7877197265625"
            y2="82.51795959472656"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="93.950439453125"
            x2="701.5409545898438"
            y1="82.51795959472656"
            y2="183.3124237060547"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="4.377518177032471"
            y="149.57069396972656"
          >
            30
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="15.077028274536133"
            x2="97.357421875"
            y1="183.6415557861328"
            y2="113.438720703125"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="97.357421875"
            x2="697.430419921875"
            y1="113.438720703125"
            y2="216.96934509277344"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="8.574006080627441"
            y="182.39324951171875"
          >
            25
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="19.187578201293945"
            x2="100.76438903808594"
            y1="215.70359802246094"
            y2="143.5676727294922"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="100.76438903808594"
            x2="693.3198852539062"
            y1="143.5676727294922"
            y2="249.83445739746094"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="12.770468711853027"
            y="214.4240264892578"
          >
            20
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="23.29814910888672"
            x2="104.17137908935547"
            y1="246.9738311767578"
            y2="172.90484619140625"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="104.17137908935547"
            x2="689.2092895507812"
            y1="172.90484619140625"
            y2="281.90777587890625"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="16.966957092285156"
            y="245.66299438476562"
          >
            15
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="27.40869903564453"
            x2="107.57836151123047"
            y1="277.4523010253906"
            y2="201.45021057128906"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="107.57836151123047"
            x2="685.0986938476562"
            y1="201.45021057128906"
            y2="313.1893005371094"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="21.163442611694336"
            y="276.11016845703125"
          >
            10
          </text>
        </g>
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="31.519269943237305"
            x2="110.9853286743164"
            y1="307.1389465332031"
            y2="229.2037811279297"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="110.9853286743164"
            x2="680.9881591796875"
            y1="229.2037811279297"
            y2="343.6790466308594"
          />
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="25.35990333557129"
            y="305.76556396484375"
          >
            5
          </text>
        </g>
        <g>
          <text
            dx="0"
            dy="3.6666666666666665"
            fill="#333"
            font-size="11"
            font-weight="normal"
            text-anchor="end"
            x="29.556392669677734"
            y="334.629150390625"
          >
            0
          </text>
        </g>
        <g>
          <polygon
            fill="#dcdcdc"
            fill-opacity="0.3"
            points="-5.475783348083496,11.45443344116211 35.62984085083008,336.0338134765625 114.3923110961914,256.1655578613281 80.3225326538086,-49.08299255371094 -5.475783348083496,11.45443344116211"
            stroke="#bfbfbf"
            stroke-opacity="1"
            stroke-width="2"
          />
        </g>
      </g>
      <g
        class="grid-block"
        transform="translate(50,100)"
      >
        <g>
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="63.029136657714844"
            x2="657.3583374023438"
            y1="307.8379211425781"
            y2="438.4094543457031"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="24.26879119873047"
            x2="63.029136657714844"
            y1="-10.297855377197266"
            y2="307.8379211425781"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="89.28329467773438"
            x2="667.6905517578125"
            y1="281.2151794433594"
            y2="405.1066589355469"
          />
          <line
            stroke="#ebebeb"
            stroke-dasharray="none"
            stroke-opacity="1"
            stroke-width="1"
            x1="52.86823654174805"
            x2="89.28329467773438"
            y1="-30.476993560791016"
            y2="281.2151794433594"
          />
          <text
            fill="#333"
            font-size="10"
            font-weight="normal"
            text-anchor="start"
            x="658.8248901367188"
            y="463.51812744140625"
          >
            MS
          </text>
          <text
            fill="#333"
            font-size="10"
            font-weight="normal"
            text-anchor="start"
            x="669.5008544921875"
            y="429.2235412597656"
          >
            Apple
          </text>
          <text
            fill="#333"
            font-size="10"
            font-weight="normal"
            text-anchor="start"
            x="679.0316162109375"
            y="396.5021057128906"
          >
            Google
          </text>
        </g>
        <g>
          <polygon
            fill="#dcdcdc"
            fill-opacity="0.3"
            points="80.3225326538086,-49.08299255371094 717.9832153320312,40.766822814941406 676.8776245117188,373.3769836425781 114.3923110961914,256.1655578613281 80.3225326538086,-49.08299255371094"
            stroke="#bfbfbf"
            stroke-opacity="1"
            stroke-width="2"
          />
        </g>
      </g>
      <g
        class="grid-panel"
        display="none"
        transform="translate(50,100)"
      >
        <rect
          fill="transparent"
          height="400"
          stroke="transparent"
          width="700"
          x="0"
          y="0"
        />
      </g>
      <g
        class="brush-polygon.line3d"
        transform="translate(50,100)"
      >
        <polygon
          fill="#FFC000"
          fill-opacity="0.6"
          points="150.39913940429688,170.66310119628906 158.4180908203125,162.89404296875 298.1420593261719,188.24456787109375 291.4499206542969,196.57028198242188 150.39913940429688,170.66310119628906"
          stroke="#4d3a00"
          stroke-opacity="0.7"
        />
        <polygon
          fill="#7977C2"
          fill-opacity="0.6"
          points="111.984130859375,341.7872619628906 120.45381927490234,332.11029052734375 257.2158203125,272.68756103515625 249.8384246826172,282.2768249511719 111.984130859375,341.7872619628906"
          stroke="#24243a"
          stroke-opacity="0.7"
        />
        <polygon
          fill="#7BBAE7"
          fill-opacity="0.6"
          points="128.43699645996094,225.77215576171875 136.75949096679688,217.26393127441406 280.20562744140625,299.1734619140625 273.3506774902344,308.6249694824219 128.43699645996094,225.77215576171875"
          stroke="#253845"
          stroke-opacity="0.7"
        />
        <polygon
          fill="#7BBAE7"
          fill-opacity="0.6"
          points="273.3506774902344,308.6249694824219 280.20562744140625,299.1734619140625 429.17254638671875,305.5605163574219 423.5818786621094,315.3968811035156 273.3506774902344,308.6249694824219"
          stroke="#253845"
          stroke-opacity="0.7"
        />
        <polygon
          fill="#FFC000"
          fill-opacity="0.6"
          points="291.4499206542969,196.57028198242188 298.1420593261719,188.24456787109375 447.6077575683594,215.36793518066406 442.242431640625,224.2503204345703 291.4499206542969,196.57028198242188"
          stroke="#4d3a00"
          stroke-opacity="0.7"
        />
        <polygon
          fill="#FFC000"
          fill-opacity="0.6"
          points="442.242431640625,224.2503204345703 447.6077575683594,215.36793518066406 606.815185546875,244.26414489746094 602.7766723632812,253.70321655273438 442.242431640625,224.2503204345703"
          stroke="#4d3a00"
          stroke-opacity="0.7"
        />
        <polygon
          fill="#7BBAE7"
          fill-opacity="0.6"
          points="423.5818786621094,315.3968811035156 429.17254638671875,305.5605163574219 585.7435302734375,368.501708984375 581.557861328125,379.1095275878906 423.5818786621094,315.3968811035156"
          stroke="#253845"
          stroke-opacity="0.7"
        />
        <polygon
          fill="#7977C2"
          fill-opacity="0.6"
          points="249.8384246826172,282.2768249511719 257.2158203125,272.68756103515625 411.72113037109375,348.07586669921875 405.7799987792969,358.5224914550781 249.8384246826172,282.2768249511719"
          stroke="#24243a"
          stroke-opacity="0.7"
        />
        <polygon
          fill="#7977C2"
          fill-opacity="0.6"
          points="405.7799987792969,358.5224914550781 411.72113037109375,348.07586669921875 579.6262817382812,316.9093933105469 574.8556518554688,327.483154296875 405.7799987792969,358.5224914550781"
          stroke="#24243a"
          stroke-opacity="0.7"
        />
      </g>
    </g>
    <g
      transform="translate(0.5,0.5)"
    />
  </svg>
</div>
`;
